solutions/usaco-991.mdx

```diff
@@ -0,0 +1,72 @@
+---
+id: usaco-991
+source: USACO Silver 2020 January
+title: Loan Repayment
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_loan_silver_jan20.html)
+
+## Implementation
+**Time Complexity:** $\mathcal{O}(\sqrt{N} \cdot \log{N})$
+
+<LanguageSection>
+  
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class LoanRepayment {
+	// our checking function in the binary search
+	static boolean isOk (long numGallons, long withinDays, long atLeast, long x) {
+		long g = 0;
+		while (withinDays > 0 && g < numGallons) {
+			long y = (numGallons - g) / x;
+			if (y < atLeast) {
+				long leftover = ((numGallons - g) + (atLeast - 1)) / atLeast;
+				return leftover <= withinDays;
+			}
+			long maxMatch = numGallons - (x * y);
+			long numDays = (maxMatch - g) / y + 1;
+			if (numDays > withinDays) {  // take a maximum
+				numDays = withinDays;
+			}
+			g += y * numDays;  // update values
+			withinDays -= numDays;
+		}
+		return g >= numGallons;
+	}
+
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio("loan");
+		long numGallons = io.nextLong();
+		long withinDays = io.nextLong();
+		long atLeast = io.nextLong();
+
+		long low = 1;
+		long high = 1000000000000L;
+
+		// keep running until we narrowed it down to one number
+		while (low < high) {
+			long mid = (low + high + 1) / 2;
+			// we can exclude everything up to but not including mid
+			if (isOk(numGallons, withinDays, atLeast, mid)) {
+				low = mid;
+			}
+			else {  // we can exclude everything including mid
```

Comment:
```suggestion
			} else {  // we can exclude everything including mid
```

Reply:
i dont do that format

Reply:
![image](https://user-images.githubusercontent.com/55369003/124362034-7b731500-dbe7-11eb-94fa-ddb5e426ad80.png)

Reply:
nahh... i think it's fine as long as it's consistent...

Reply:
i mean nathan and michael said no so...

Reply:
i think they were just joking... right? 👀 @thecodingwizard @caoash

Reply:
![image](https://user-images.githubusercontent.com/63619830/124366524-4f668c80-dc05-11eb-8af8-aaf151dc413a.png)

Reply:
still looks pretty ugly imo

Reply:
![image](https://user-images.githubusercontent.com/63619830/124366626-0105bd80-dc06-11eb-8967-bb82229efc2d.png)

Reply:
what's wrong w/ enforcing it though

Reply:
nothing is, but it doesn't matter so the author can choose.

Reply:
ok lmao